numba/np/npyimpl.py

```diff
@@ -383,7 +383,17 @@ def numpy_ufunc_kernel(context, builder, sig, args, ufunc, kernel_class):
     # assume outputs are all the same size, which numpy requires
 
     loopshape = outputs[0].shape
-    with cgutils.loop_nest(builder, loopshape, intp=intpty) as loop_indices:
+
+    input_layouts = [inp.layout for inp in inputs if isinstance(inp, _ArrayHelper)]
+    num_c_layout = len([x for x in input_layouts if x == 'C'])
+    num_f_layout = len([x for x in input_layouts if x == 'F'])
```

Comment:
TODO check what numpy does

Reply:
There are some information around https://github.com/numpy/numpy/blob/ff4b33edcf6819f24f9a1b4a483d64873cea2d5c/numpy/core/src/common/lowlevel_strided_loops.h#L621.

Numpy will perform a trivial loop iteration if arguments are all C or F. Otherwise, it will use the array iterator which we don't have. 

The logic here is doing the best effort. If more arrays are in F layout then in C layout, we will use that F iteration order; otherwise, we will use C order by default.

Reply:
Perhaps not for this PR, but maybe sometime... given `ndim` is known for each array, if there are 0D arrays should they get special treatment? e.g. imagine `input_layouts` evaluated as `['F', 'C', 'C', 'C', 'C']`, but the `'C'` arrays were all `0D` and the `'F'` array was `>1D`, it'd be most efficient to iterate in `'F'` order?

Reply:
You are right. The NumPy code has logic for array scalars.

Reply:
#8618 tracks.